src/app/components/ParticipatedContests/index.js

```diff
@@ -1,116 +1,95 @@
 import PropTypes from 'prop-types';
+import { useHistory } from 'react-router-dom';
 
-import { get } from 'src/utils/fetchUtils';
+// Components
+import Table from 'src/app/common-ui/Table';
 
-const pastContestsPerPage = 10;
-let numberOfContests = 0;
+// Utils
+import { makeUrl } from 'src/utils/url';
+import styled from 'styled-components';
 
-// This function lets us know the range of less than 6 indexes, denoting the pages next to that one users 're handling. In which, the page user 're handling is in the middle.
-function splitPages(numberOfItems, itemsPerPage, page) {
-  if (Math.ceil(numberOfItems / itemsPerPage) <= 5) {
-    return [1, Math.ceil(numberOfItems / itemsPerPage)];
-  } else {
-    const [first, last] = [1, Math.ceil(numberOfItems / itemsPerPage)];
-    let [from, to] = [page - 2, page + 2];
-    if (to > last) {
-      from -= to - last;
-      to = last;
-    } else if (from < first) {
-      to += first - from;
-      from = first;
-    }
-    return [from, to];
-  }
-}
+// Apis
+import { apiGetParticipations } from 'src/api';
+import Skeleton from 'src/app/common-ui/Skeleton';
 
-function ParticipatedContests({ token, username }) {
-  const [page, setPage] = useState(1);
-  const [pastContests, setPastContests] = useState([]);
-  useEffect(() => {
-    get(
-      /*
-      `https://test.api.freecontest.net/api/v1/participations?username=${username}&offset=${
-        (page - 1) * pastContestsPerPage
-      }&limit=${pastContestsPerPage}`, */
-      'https://my-json-server.typicode.com/upi05/mock_API_for_Fc5y/participations',
-      {
-        headers: {
-          Authorization: `Bearer ${token}`,
-        },
-      },
-      true,
-    ).then((res) => {
-      // Don't forget to handle errors
-      numberOfContests = res.data.data.total;
-      setPastContests(res.data.data.participations);
+// Contants
+import { TB_CONFIG } from './config';
+import { ROUTE_CONTEST } from 'src/app/routes/constants';
+
+// Styles
+const ContestTitle = styled.h1`
+  font-weight: 700;
+  font-size: 16px;
+  color: var(--primary-default);
+  cursor: pointer;
+`;
+
+let totalContests = 0;
+
+function ParticipatedContests({ username }) {
+  const [tableConfig, setTableConfig] = useState(TB_CONFIG);
+  const [currentPage, setCurrentPage] = useState(1);
+  const [currentPageSize, setCurrentPageSize] = useState(10);
+  const history = useHistory();
+
+  // Helper function to format table data
+  const formatTableData = (data) => {
+    return data.map((d, i) => {
+      return {
+        index: i + 1 + (currentPage - 1) * currentPageSize,
+        contestName: (
+          <ContestTitle
+            onClick={() => history.push(makeUrl(ROUTE_CONTEST, { contestName: d.contest_name }))}
+          >
+            {d.contest_title || ''}
+          </ContestTitle>
+        ),
+        ranking: `${d.rank_in_contest}/${d.contest_total_participations}`,
+        rating: (
+          <>
+            {d.rating}
+            <span
```

Comment:
missing spacing between rating and rating_change, should add style padding-left for this span